### ISSUE TITLE
If-None-Match: "" sent when --etag-save and --etag-compare used together

### ISSUE THREAD

**Issue by simonw** [curl/curl#5309]
### I did this
I tried using `curl -vv --etag-save etag.txt --etag-compare etag.txt ...` twice - but the file was downloaded every time. On close inspection of the verbose output the `If-None-Match` header was set to `""` for both requests - when on the second request it should have been set to the value that was saved in the `etag.txt` file.
```
$ curl -vv --etag-save etag.txt --etag-compare etag.txt https://c19downloads.azureedge.net/downloads/data/data_latest.json -o data_latest.json

$ ls # demonstrating directory is empty

$ curl -vv --etag-save etag.txt --etag-compare etag.txt https://c19downloads.azureedge.net/downloads/data/data_latest.json -o data_latest.json
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 13.107.246.10:443...
* Connected to c19downloads.azureedge.net (13.107.246.10) port 443 (#0)
* ALPN, offering http/1.1
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate: *.azureedge.net
* Server certificate: Microsoft IT TLS CA 1
* Server certificate: Baltimore CyberTrust Root
> GET /downloads/data/data_latest.json HTTP/1.1
> Host: c19downloads.azureedge.net
> User-Agent: curl/7.69.1
> Accept: */*
> If-None-Match: ""
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Length: 1672673
< Content-Type: application/json
< Content-MD5: OJP5Gcz41pLNsqwdqkPYuA==
< Last-Modified: Tue, 28 Apr 2020 18:14:29 GMT
< Accept-Ranges: bytes
< ETag: "0x8D7EB9FFE506DA1"
< x-ms-request-id: 8514ea4a-301e-0054-55b5-1d946d000000
< x-ms-version: 2014-02-14
< x-ms-meta-AzureWebJobsParentId: 02332584-c8e0-43db-8d74-de8380ba8602
< x-ms-lease-status: unlocked
< x-ms-lease-state: available
< x-ms-blob-type: BlockBlob
< Access-Control-Expose-Headers: x-ms-request-id,Server,x-ms-version,x-ms-meta-AzureWebJobsParentId,Content-Type,Last-Modified,ETag,Content-MD5,x-ms-lease-status,x-ms-lease-state,x-ms-blob-type,Accept-Ranges,Content-Length,Date,Transfer-Encoding
< Access-Control-Allow-Origin: *
< X-Azure-Ref: 0sL2oXgAAAADy6roS3RGtToh7+odykyddU0pDRURHRTA1MjEAZjU2NzRjOGEtNzE4Ny00ZGIwLWEyNjYtZTE5YThiZGJmMzY4
< Date: Tue, 28 Apr 2020 23:35:12 GMT
< 
{ [3234 bytes data]
100 1633k  100 1633k    0     0   634k      0  0:00:02  0:00:02 --:--:--  634k
* Connection #0 to host c19downloads.azureedge.net left intact

$ ls
data_latest.json	etag.txt

$ cat etag.txt
0x8D7EB9FFE506DA1

# This should use the etag from that file - but it sends If-None-Match: ""
$ curl -vv --etag-save etag.txt --etag-compare etag.txt https://c19downloads.azureedge.net/downloads/data/data_latest.json -o data_latest.json
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 13.107.246.10:443...
* Connected to c19downloads.azureedge.net (13.107.246.10) port 443 (#0)
* ALPN, offering http/1.1
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate: *.azureedge.net
* Server certificate: Microsoft IT TLS CA 1
* Server certificate: Baltimore CyberTrust Root
> GET /downloads/data/data_latest.json HTTP/1.1
> Host: c19downloads.azureedge.net
> User-Agent: curl/7.69.1
> Accept: */*
> If-None-Match: ""
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Length: 1672673
< Content-Type: application/json
< Content-MD5: OJP5Gcz41pLNsqwdqkPYuA==
< Last-Modified: Tue, 28 Apr 2020 18:14:29 GMT
< Accept-Ranges: bytes
< ETag: "0x8D7EB9FFE506DA1"
< x-ms-request-id: 9d0abb3e-801e-0023-40b5-1d41f9000000
< x-ms-version: 2014-02-14
< x-ms-meta-AzureWebJobsParentId: 02332584-c8e0-43db-8d74-de8380ba8602
< x-ms-lease-status: unlocked
< x-ms-lease-state: available
< x-ms-blob-type: BlockBlob
< Access-Control-Expose-Headers: x-ms-request-id,Server,x-ms-version,x-ms-meta-AzureWebJobsParentId,Content-Type,Last-Modified,ETag,Content-MD5,x-ms-lease-status,x-ms-lease-state,x-ms-blob-type,Accept-Ranges,Content-Length,Date,Transfer-Encoding
< Access-Control-Allow-Origin: *
< X-Azure-Ref: 0tr2oXgAAAADi99T06YVqRID7WImJRKhKU0pDRURHRTA1MTcAZjU2NzRjOGEtNzE4Ny00ZGIwLWEyNjYtZTE5YThiZGJmMzY4
< Date: Tue, 28 Apr 2020 23:35:18 GMT
< 
{ [3234 bytes data]
100 1633k  100 1633k    0     0   600k      0  0:00:02  0:00:02 --:--:--  600k
* Connection #0 to host c19downloads.azureedge.net left intact
```
Here's the thing: if I run the command again just using `--etag-compare` it works as expected - the `If-None-Match` header is set to the correct string:
```
$ cat etag.txt 
0x8D7EB9FFE506DA1

$ curl -vv --etag-compare etag.txt https://c19downloads.azureedge.net/downloads/data/data_latest.json -o data_latest.json                     
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 13.107.246.10:443...
* Connected to c19downloads.azureedge.net (13.107.246.10) port 443 (#0)
* ALPN, offering http/1.1
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate: *.azureedge.net
* Server certificate: Microsoft IT TLS CA 1
* Server certificate: Baltimore CyberTrust Root
> GET /downloads/data/data_latest.json HTTP/1.1
> Host: c19downloads.azureedge.net
> User-Agent: curl/7.69.1
> Accept: */*
> If-None-Match: "0x8D7EB9FFE506DA1"
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 304 The condition specified using HTTP conditional header(s) is not met.
< Content-Type: application/json
< x-ms-request-id: fc5bc254-101e-0053-03b6-1df80e000000
< x-ms-version: 2014-02-14
< Access-Control-Expose-Headers: x-ms-request-id,Server,x-ms-version,Content-Length,Date,Transfer-Encoding
< Access-Control-Allow-Origin: *
< X-Azure-Ref: 0WL6oXgAAAAAerlWGaDM7R5al7kBCQcaZU0pDRURHRTA1MTYAZjU2NzRjOGEtNzE4Ny00ZGIwLWEyNjYtZTE5YThiZGJmMzY4
< Date: Tue, 28 Apr 2020 23:38:01 GMT
< 
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
* Connection #0 to host c19downloads.azureedge.net left intact
```
So the bug only appears to occur when `--etag-save` and `--etag-compare` are used together.

### I expected the following

The second time I ran the `curl -vv --etag-save etag.txt --etag-compare etag.txt ...` command the `If-Modified-Since` header should have been correctly populated with the contents of the `etag.txt` file.

### curl/libcurl version
```
$ curl --version
curl 7.69.1 (x86_64-apple-darwin19.3.0) libcurl/7.69.1 SecureTransport zlib/1.2.11
Release-Date: 2020-03-11
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IPv6 Largefile libz NTLM NTLM_WB SSL UnixSockets
```

### operating system
macOS:
```
$ uname -a
Darwin Simons-MacBook-Pro.local 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
```

**Comment by simonw** [comment]
This should have been addressed by #4669 and #4678 - which were implemented in https://github.com/curl/curl/commit/9ea769e15f60f8bebf42bffd708658296a812d6e. That commit is part of the `7.68.1` tag which is the version I am using.

**Comment by simonw** [comment]
I tried this using a freshly compiled binary on Ubuntu 18:04 and got the same result.

**Comment by simonw** [comment]
I can also replicate this bug using the example from #4669:
```
$ curl https://curl.haxx.se/ --etag-compare moo -v --etag-save moo -o file
$ curl https://curl.haxx.se/ --etag-compare moo -v --etag-save moo -o file
```
Running that twice also appears not to correctly populate the `If-None-Match` header on the second execution.

**Comment by hubgit** [comment]
Looking at the test in https://github.com/curl/curl/pull/4678/files it seems to work if `--etag-compare` and `--etag-save` refer to different files, so this would be a workaround:

```sh
touch moo
curl https://curl.haxx.se/ --etag-compare moo -v --etag-save moo-tmp -o file
mv moo-tmp moo
curl https://curl.haxx.se/ --etag-compare moo -v --etag-save moo-tmp -o file
```

**Comment by bagder** [comment]
Sounds like the fix in #5180 might be what you want. Try 7.70.0, released today!

**Comment by simonw** [comment]
I just tried it out in `curl 7.70.0` (installed on macOS via `brew upgrade curl`) and it worked!

```
$ curl -vv --etag-save etag.txt --etag-compare etag.txt https://c19downloads.azureedge.net/downloads/data/data_latest.json -o data_latest.json

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 13.107.246.10:443...
* Connected to c19downloads.azureedge.net (13.107.246.10) port 443 (#0)
* ALPN, offering http/1.1
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate: *.azureedge.net
* Server certificate: Microsoft IT TLS CA 1
* Server certificate: Baltimore CyberTrust Root
> GET /downloads/data/data_latest.json HTTP/1.1
> Host: c19downloads.azureedge.net
> User-Agent: curl/7.70.0
> Accept: */*
> If-None-Match: ""
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Length: 1707560
< Content-Type: application/json
< Content-MD5: zksf48a28UunN1kz1GplPQ==
< Last-Modified: Wed, 29 Apr 2020 16:31:05 GMT
< Accept-Ranges: bytes
< ETag: "0x8D7EC5AB6D987A3"
< x-ms-request-id: cb66853e-301e-0044-3b49-1e5105000000
< x-ms-version: 2014-02-14
< x-ms-meta-AzureWebJobsParentId: 1ad5c66b-cc50-4c52-bc91-6b7a216ec370
< x-ms-lease-status: unlocked
< x-ms-lease-state: available
< x-ms-blob-type: BlockBlob
< Access-Control-Expose-Headers: x-ms-request-id,Server,x-ms-version,x-ms-meta-AzureWebJobsParentId,Content-Type,Last-Modified,ETag,Content-MD5,x-ms-lease-status,x-ms-lease-state,x-ms-blob-type,Accept-Ranges,Content-Length,Date,Transfer-Encoding
< Access-Control-Allow-Origin: *
< X-Azure-Ref: 0Q7WpXgAAAACH6I6xH9FxSoHwwr203F2WU0pDRURHRTA1MTEAZjU2NzRjOGEtNzE4Ny00ZGIwLWEyNjYtZTE5YThiZGJmMzY4
< Date: Wed, 29 Apr 2020 17:11:31 GMT
< 
{ [3234 bytes data]
100 1667k  100 1667k    0     0  1740k      0 --:--:-- --:--:-- --:--:-- 1740k
* Connection #0 to host c19downloads.azureedge.net left intact
$ ls

data_latest.json	etag.txt
$ cat etag.txt 

0x8D7EC5AB6D987A3
$ curl -vv --etag-save etag.txt --etag-compare etag.txt https://c19downloads.azureedge.net/downloads/data/data_latest.json -o data_latest.json

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 13.107.246.10:443...
* Connected to c19downloads.azureedge.net (13.107.246.10) port 443 (#0)
* ALPN, offering http/1.1
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate: *.azureedge.net
* Server certificate: Microsoft IT TLS CA 1
* Server certificate: Baltimore CyberTrust Root
> GET /downloads/data/data_latest.json HTTP/1.1
> Host: c19downloads.azureedge.net
> User-Agent: curl/7.70.0
> Accept: */*
> If-None-Match: "0x8D7EC5AB6D987A3"
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 304 The condition specified using HTTP conditional header(s) is not met.
< Content-Type: application/json
< x-ms-request-id: a163cbb8-c01e-0022-1849-1e1e25000000
< x-ms-version: 2014-02-14
< Access-Control-Expose-Headers: x-ms-request-id,Server,x-ms-version,Content-Length,Date,Transfer-Encoding
< Access-Control-Allow-Origin: *
< X-Azure-Ref: 0WrWpXgAAAACKKz6wgFsYS7BMTZZKiv/gU0pDRURHRTA1MTYAZjU2NzRjOGEtNzE4Ny00ZGIwLWEyNjYtZTE5YThiZGJmMzY4
< Date: Wed, 29 Apr 2020 17:11:54 GMT
< 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
* Connection #0 to host c19downloads.azureedge.net left intact
$ 
```
